tests/test_models.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+from django.contrib.sites.models import Site
+
+from controlled_vocabularies.models import Vocabulary, Term, Property
+
+pytestmark = pytest.mark.django_db
+
+
+def create_vocab(name='languages', label='Language', order='name',
+                 maintainer='Buddy Noone', maintainerEmail='noone@nowhere.com',
+                 definition='Lorem ipsum dolor sit amet'):
+    """Creates a saved instance of the Vocabulary model."""
+    vocab_instance = Vocabulary(name=name,
+                                label=label,
+                                order=order,
+                                maintainer=maintainer,
+                                maintainerEmail=maintainer,
+                                definition=definition)
+    vocab_instance.save()
+    return(vocab_instance)
+
+
+def create_term(name='spanish', label='Spanish', order=2):
+    """Creates a saved instance of the Term model."""
+    vocab_instance = create_vocab()
+    term_instance = Term(vocab_list=vocab_instance,
+                         name=name,
+                         label=label,
+                         order=order)
+    term_instance.save()
+    return(term_instance)
+
+
+def create_property(property_name='dialect', label='Catalan'):
+    """Creates a saved instance of the Property model."""
+    term_instance = create_term()
+    property_instance = Property(term_key=term_instance,
+                                 property_name=property_name,
+                                 label=label)
+    property_instance.save()
+    return(property_instance)
+
+
+class TestVocabulary:
+    def test_save(self):
+        assert Vocabulary.objects.all().count() == 0
+        create_vocab()
+        assert Vocabulary.objects.all().count() == 1
+
+    def test_unicode(self):
+        vocab_instance = create_vocab()
+        assert unicode(vocab_instance) == 'languages'
+
+    @pytest.mark.xfail(reason='Input is not being cleaned before being saved.')
+    def test_whitespace_stripped(self):
+        create_vocab(
+            name=' spaces  here ',
+            label=' Spaces here   too  ',
+            maintainer=' Please   Fix Me ',
+            maintainerEmail='  here@there.com ',
+            definition='  Strip leading   and trailing, condense   others. '
+        )
+        vocab_from_db = Vocabulary.objects.get(pk=1)
+
+        assert vocab_from_db.name == 'spaces here'
+        assert vocab_from_db.label == 'Spaces here too'
+        assert vocab_from_db.maintainer == 'Please Fix Me'
+        assert vocab_from_db.maintainerEmail == 'here@there.com'
+        assert vocab_from_db.definition == 'Strip leading and trailing, condense others.'
+
+
+class TestTerm:
+    def test_unicode(self):
+        term_instance = create_term()
+        assert unicode(term_instance) == 'spanish'
+
+    def test_save(self):
+        assert Term.objects.all().count() == 0
+        create_term()
```

Comment:
But we still won't be calling the SUT directly from the test which is the issue that I have with it. We will be relying on the implementation of `create_term` to do that for us.